### ISSUE TITLE
Model collapse after adding encoder layers.

### ISSUE THREAD

**Issue by neuer93** [oliverguhr/transformer-time-series-prediction#15]
Hi, thanks for providing the code.

In `transformer-singlestep.py`, I change `model = TransAm().to(device)` to `model = TransAm(num_layers=5).to(device)`. When I train the model, I find the model outputs same value (0) for all inputs. I thought that increasing the number of layers can make model more expressive, but it results in worse performance.

Do you meet the same problem before?  I am not sure if I need to change the training setting.

(BTW, I also tried suggestion in another issue. I remove `self.` in `self.encoder_layer`, but still can't make training converge.)

Prediction for epoch 100:

![image](https://user-images.githubusercontent.com/4123046/122647047-50b99480-d0f0-11eb-8d82-83a997a18a2f.png)

**Comment by oliverguhr** [comment]
Thanks for reporting this issue. I had similar issues with this simple sin wave data too. You can try to increase the learning rate or train on more data. But basically, one or two layers are enough for this super simple toy sample - so more layers won't improve the results.

**Comment by oliverguhr** [comment]
Feel free to reopen this issue if you have any questions.
